operator/app/pom.xml

```diff
@@ -263,13 +262,17 @@
                 </executions>
             </plugin>
             <plugin>
-                <artifactId>maven-failsafe-plugin</artifactId>
-                <version>${maven-failsafe-plugin.version}</version>
+                <artifactId>maven-surefire-plugin</artifactId>
                 <executions>
+                <!--
+                    need to run at "verify" to support testing with images (which are built at "package")
+                    using surefire instead of failsafe to align with rest of the KC project
+                -->
                     <execution>
+                        <id>default-test</id>
+                        <phase>verify</phase>
```

Comment:
For future reference, I don't have any better proposal now.
But this approach to running all of the tests in the `verify` phase compromises the speed of running plain unit tests and we might want to re-iterate in the future.

Reply:
Yes, and not only unit tests unfortunately. Even the integration tests could be run at `test` phase. Basically, we need `verify` only when testing in the "remote" mode (operator deployed in the cluster). Maybe we could introduce some special profile for this scenario.